src/components/TableOfContents/TableOfContentsLink.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react"
+import { Link as GatsbyLink } from "gatsby"
+import { SystemStyleObject, cssVar } from "@chakra-ui/react"
+import Link from "../Link"
+import { getCustomId, Item, trimmedTitle } from "./tableOfContents.utils"
+
+export interface IPropsTableOfContentsLink {
+  depth: number
+  item: Item
+  activeHash?: string
+}
+
+const TableOfContentsLink: React.FC<IPropsTableOfContentsLink> = ({
+  depth,
+  item,
+  activeHash,
+}) => {
+  const url = `#${getCustomId(item.title)}`
+  const isActive = activeHash === url
+  const isNested = depth === 2
+  let classes = ""
+  if (isActive) {
+    classes += " active"
+  }
+  if (isNested) {
+    classes += " nested"
+  }
+
+  const $dotBg = cssVar("dot-bg")
+
+  const hoverOrActiveStyle: SystemStyleObject = {
+    color: "primary",
+    _after: {
+      content: `""`,
+      background: $dotBg.reference,
+      border: "1px",
+      borderColor: "primary",
+      borderRadius: "50%",
+      boxSize: 2,
+      position: "absolute",
+      left: "-1.29rem",
+      top: "50%",
+      mt: -1,
+    },
+  }
+
+  return (
+    <Link
+      as={GatsbyLink}
+      to={url}
+      className={classes}
+      textDecoration="none"
+      display="inline-block"
+      position="relative"
+      color="textTableOfContents"
+      mb="0.5rem !important"
+      width={{ base: "100%", lg: "auto" }}
+      _hover={{
+        ...hoverOrActiveStyle,
+      }}
+      sx={{
+        [$dotBg.variable]: "colors.background",
+        "&.active": {
+          [$dotBg.variable]: "colors.primary",
+          ...hoverOrActiveStyle,
+        },
+        "&.nested": {
+          _before: {
+            content: `"⌞"`,
+            opacity: 0.5,
+            display: "inline-flex",
+            position: "absolute",
+            left: "-14px",
+            top: -1,
+          },
+          "&.active, &:hover": {
+            _after: {
+              left: "-2.29rem",
+            },
+          },
+        },
+      }}
+    >
+      {trimmedTitle(item.title)}
+    </Link>
+  )
+}
+
+export default TableOfContentsLink
```

Comment:
Lets remove the `TableOfContents` prefix now that we have a folder with that name. Rename the file as well.
```suggestion
export default Link
```